remove Cylc 7 compatibility mode (#7325)
cylc/cylc-flow: changes.d/6850.break.md

> The version of Jinja2 has been upgraded from 3.0 to 3.1, the version at which deprecated language features were removed.

Reviewer: ```suggestion
The version of Jinja2 has been upgraded from 3.0 to 3.1, removing deprecated language features. See https://jinja.palletsprojects.com/en/stable/changes/#version-3-1-0 for details.
```
